File: src/leetcode/00001.two-sum/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const map = new Map(nums.map((item, index) => [item, index]));
  for (let i = 0; i < nums.length; i++) {
    const index = map.get(target - nums[i]);
    if (index !== undefined && index !== i) {
      return [i, map.get(target - nums[i])];
    }
  }
};

console.log(twoSum([2, 7, 11, 15], 9));

File: src/leetcode/00001.two-sum/1.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    for i, x := range nums {
        for j := i + 1; j < len(nums); j++ {
            if x+nums[j] == target {
                return []int{i, j}
            }
        }
    }
    return nil
}

func main() {
    fmt.Print(twoSum([]int{2, 7, 11, 15}, 9))
}
